### PR TITLE
Move Runner to typescript

### PR DESCRIPTION
**Pull request by FriggaHel**
Convert Runner to use TypeScript.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,6 +47,7 @@
     "dotenv": "16.0.3",
     "fluent-ffmpeg": "^2.1.2",
     "lodash": "4.17.21",
+    "mkdirp": "^3.0.0",
```
Comment (alexplischke) [inline package.json:50]: Technically a devDependency, no?

**Comment by alexplischke** [comment]
Regarding the `tsconfig.json`: maybe we should make use of `extends`? I used it for our [testcomposer lib](https://github.com/saucelabs/node-testcomposer/blob/main/tsconfig.json#L2).

Here's a [list of available presets](https://github.com/tsconfig/bases/tree/main/bases).

**Review by alexplischke — COMMENTED** [review]
Bump typescript to 5.x.x?

File: package.json (hunk 1 above)
Comment (FriggaHel) [reply package.json:50]: This is used in `custom-reporter.ts` to write the files in the folders associated to a suite. Not a `devDependency`.

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+export type XmlTestCaseContainer = {
+  testcase: XmlTestCase[];
+};
+
+export type XmlSuite = {
+  testsuite?: (XmlSuiteAttrContainer | XmlProperties | XmlTestCaseContainer)[];
+} & XmlSuiteAttrContainer;
+
+export type HasXmlSuite = {
```
Comment (alexplischke) [inline src/types.ts:105]: Maybe more apt?
```suggestion
export type XmlSuiteContainer = {
```

another variant is
```suggestion
export type XmlSuiteHolder = {
```

`Is/Has` inches a bit too close to methods :x

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -30,6 +30,9 @@ jobs:
       - name: Install Dependencies
         run: npm ci
 
+      - name: Build Project
+        run: npm run build
+
```
Comment (alexplischke) [inline .github/workflows/test.yml:35]: What about https://github.com/saucelabs/sauce-cypress-runner/blob/main/.github/workflows/release.yml ? Doesn't the release pipeline need to build as well now before packaging?

Comment (FriggaHel) [reply .github/workflows/test.yml:35]: This part is deal with in the `bundle.sh` script: https://github.com/saucelabs/sauce-cypress-runner/blob/DEVX-2238-move-to-typescript/scripts/bundle.sh#L17

**Review by alexplischke — APPROVED** [review]

**Pull request merged**